plugins/inputs/influxdb_listener/influxdb_listener.go

```diff
@@ -0,0 +1,404 @@
+package influxdb_listener
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	tlsint "github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/selfstat"
+)
+
+const (
+	// defaultMaxBodySize is the default maximum request body size, in bytes.
+	// if the request body is over this size, we will return an HTTP 413 error.
+	defaultMaxBodySize = 32 * 1024 * 1024
+)
+
+type InfluxDBListener struct {
+	ServiceAddress string `toml:"service_address"`
+	port           int
+	tlsint.ServerConfig
+
+	ReadTimeout   internal.Duration `toml:"read_timeout"`
+	WriteTimeout  internal.Duration `toml:"write_timeout"`
+	MaxBodySize   internal.Size     `toml:"max_body_size"`
+	BasicUsername string            `toml:"basic_username"`
+	BasicPassword string            `toml:"basic_password"`
+	DatabaseTag   string            `toml:"database_tag"`
+
+	timeFunc influx.TimeFunc
+
+	listener net.Listener
+	server   http.Server
+
+	acc telegraf.Accumulator
+
+	bytesRecv       selfstat.Stat
+	requestsServed  selfstat.Stat
+	writesServed    selfstat.Stat
+	queriesServed   selfstat.Stat
+	pingsServed     selfstat.Stat
+	requestsRecv    selfstat.Stat
+	notFoundsServed selfstat.Stat
+	buffersCreated  selfstat.Stat
+	authFailures    selfstat.Stat
+
+	log telegraf.Logger
+
+	mux http.ServeMux
+}
+
+const sampleConfig = `
+  ## Address and port to host InfluxDB listener on
+  service_address = ":8186"
+
+  ## maximum duration before timing out read of the request
+  read_timeout = "10s"
+  ## maximum duration before timing out write of the response
+  write_timeout = "10s"
+
+  ## Maximum allowed HTTP request body size in bytes.
+  ## 0 means to use the default of 32MiB.
+  max_body_size = "32MiB"
+
+  ## Optional tag name used to store the database. 
+  ## If the write has a database in the query string then it will be kept in this tag name.
+  ## This tag can be used in downstream outputs.
+  ## The default value of nothing means it will be off and the database will not be recorded.
+  # database_tag = ""
+
+  ## Set one or more allowed client CA certificate file names to
+  ## enable mutually authenticated TLS connections
+  tls_allowed_cacerts = ["/etc/telegraf/clientca.pem"]
+
+  ## Add service certificate and key
+  tls_cert = "/etc/telegraf/cert.pem"
+  tls_key = "/etc/telegraf/key.pem"
+
+  ## Optional username and password to accept for HTTP basic authentication.
+  ## You probably want to make sure you have TLS configured above for this.
+  # basic_username = "foobar"
+  # basic_password = "barfoo"
+`
+
+func (h *InfluxDBListener) SampleConfig() string {
+	return sampleConfig
+}
+
+func (h *InfluxDBListener) Description() string {
+	return "InfluxDB listener"
```

Comment:
Let's come up with a better description.

Reply:
What's the preferred format?  I chose this after looking at the other *_listener plugin descriptions which are similar.

I looked over the other plugin descriptions and couldn't find a clear pattern.  For service inputs they generally describe what the plugin is, but for non-service inputs they generally describe what the plugin does.  For example, socket listener says "Generic socket listener capable of handling multiple socket types" and memory says "Read metrics about memory usage".  Some of the "does" style ones are first person "read" and some are third person "reads".  There are a few that are long complete sentences "This plugin implements x to gather y data to troubleshoot z".

Anyway, do either of these sound better or do you have suggestions?
"HTTP server implementing InfluxDB 1.x API to accept metrics"
"Accept metrics over InfluxDB 1.x HTTP API"

Reply:
Either of those sound great, and I love if we could decide on and write down some guidelines to ease writing descriptions in the future.

Reply:
I've changed it to "Accept metrics over InfluxDB 1.x HTTP API".  I agree we should get docs to review all the descriptions.